### PR TITLE
[stable/promethus-mongodb-exporter] Customize serviceAccountName for mongodb-exporter deployment

### PR DESCRIPTION
**Pull request by nitinmohan87**
@steven-sheehy 

Signed-off-by: Nitin Mohan <nitinmohan87@gmail.com>

#### Is this a new chart
No

#### What this PR does / why we need it:
This PR adds a customizable `serviceAccountName` field for the prometheus-mongodb-exporter chart. This makes it possible to create mongodb-exporter pods in custom service accounts.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

**Comment by maorfr** [comment]
check this out: https://helm.sh/docs/chart_best_practices/#role-based-access-control

File: stable/prometheus-mongodb-exporter/Chart.yaml [hunk 1]
```diff
@@ -13,4 +13,4 @@ maintainers:
 name: prometheus-mongodb-exporter
 sources:
 - https://github.com/percona/mongodb_exporter
-version: 2.2.1
+version: 2.2.2
```
Comment (steven-sheehy) [inline stable/prometheus-mongodb-exporter/Chart.yaml]: `2.3.0` for new features

File: stable/prometheus-mongodb-exporter/README.md [hunk 2]
```diff
@@ -51,6 +51,7 @@ podAnnotations:
 | `resources` | Pod resource requests and limits | `{}` |
 | `env` | Extra environment variables passed to pod | `{}` |
 | `securityContext` | Security context for the pod | See values.yaml |
+| `serviceAccountName` | Service account name for the pod | `default` |
```
Comment (steven-sheehy) [inline stable/prometheus-mongodb-exporter/README.md]: Most charts don't attach outside created service accounts to a deployment. You should instead include the `ServiceAccount`  yaml in the chart using the best practices that maorfr mentioned.

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: stable/prometheus-mongodb-exporter/README.md (hunk 2 above)
Comment (nitinmohan87) [reply stable/prometheus-mongodb-exporter/README.md]: Thank you @steven-sheehy and @maorfr . I have added a commit to adhere to the best practices.

**Comment by nitinmohan87** [comment]
@steven-sheehy @maorfr Could you please review this when you get a chance and let me know if anything needs addressing? Would be nice to use a released version of this chart.

File: stable/prometheus-mongodb-exporter/templates/role.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{{- if .Values.rbac.create }}
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: Role
+metadata:
+  name: {{ template "prometheus-mongodb-exporter.fullname" . }}
+  labels:
+    app: {{ template "prometheus-mongodb-exporter.name" . }}
```
Comment (steven-sheehy) [inline stable/prometheus-mongodb-exporter/templates/role.yaml]: All of the new files you added use the wrong labels. Please copy the labels from the `deployment.yaml`.

File: stable/prometheus-mongodb-exporter/templates/role.yaml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{{- if .Values.rbac.create }}
+apiVersion: rbac.authorization.k8s.io/v1beta1
```
Comment (steven-sheehy) [inline stable/prometheus-mongodb-exporter/templates/role.yaml]: `rbac.authorization.k8s.io/v1`

File: stable/prometheus-mongodb-exporter/templates/role.yaml [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+kind: Role
+metadata:
+  name: {{ template "prometheus-mongodb-exporter.fullname" . }}
+  labels:
+    app: {{ template "prometheus-mongodb-exporter.name" . }}
+    chart: {{ .Chart.Name }}-{{ .Chart.Version }}
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+{{- end }}
```
Comment (steven-sheehy) [inline stable/prometheus-mongodb-exporter/templates/role.yaml]: What's the point of this Role? It has no `rules` section, making it useless. I don't see the point of the two rbac files you added or `rbac.create`. I would suggest you remove them and just keep the serviceAccount logic.

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

**Comment by nitinmohan87** [comment]
Thanks for the review @steven-sheehy . Added a commit to address your suggestions.

**Comment by steven-sheehy** [comment]
/lgtm
Thanks

**Pull request merged**